### PR TITLE
Return InvalidArgument when can't create backup uri

### PR DESCRIPTION
**Pull request by hsadoyan**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Currently we're getting SLO noise because invalid URIs are being treated as server errors
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Handle CreateBackupURI errors as InvalidArgument 
```

### CODE REVIEW

File: pkg/csi_driver/controller.go [hunk 1]
```diff
@@ -868,9 +868,11 @@ func (s *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateSn
 	}
 
 	// Check for existing snapshot
-	backupUri, _, err := file.CreateBackupURI(filer, req.Name, util.GetBackupLocation(req.GetParameters()))
+	backupLocation := util.GetBackupLocation(req.GetParameters())
+	backupUri, _, err := file.CreateBackupURI(filer, req.Name, backupLocation)
 	if err != nil {
-		return nil, err
+		klog.Errorf("Failed to create backup URI from given name %s and location %s, error: %v", req.Name, backupLocation, err.Error())
+		return nil, status.Error(codes.InvalidArgument, err.Error())
```
Comment (leiyiz) [inline pkg/csi_driver/controller.go:875]: My question is that: are ALL errors produced by `CreateBackupURI` invalidArgument?

Comment (saikat-royc) [reply pkg/csi_driver/controller.go:875]: Given that BackupURI takes the name, and the params, its fair to mark it as invalidArg (unless we have a bug in parsing logic itself).

**Review by leiyiz — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**